includes/class-freemius.php

```diff
@@ -6387,6 +6479,23 @@ function opt_in(
 		) {
 			$this->_logger->entrance();
 
+			if ( ! $is_uninstall && fs_request_is_action( $this->_slug . '_opt_in' ) ) {
+				$site = $this->get_api_site_scope()->call( '/', 'put', array(
+					'is_disconnected' => false
+				) );
+
+				if ( ! $this->is_api_error( $site ) ) {
+					$this->_site = new FS_Site( $site );
+					$this->_store_site();
+
+					$this->schedule_sync_cron();
+
+					fs_redirect( $this->get_after_activation_url( 'after_connect_url' ) );
```

Comment:
@fajardoleo don't redirect after reconnecting, instead, add an admin notice on the same page. You can use the same string from the `opt-out-message-appreciation` message.